### DISCUSSION TITLE
Help needed with generating database with different fields

### DISCUSSION THREAD

**Discussion by nickshanks347** [fastapi-users/fastapi-users#1030] [Q&A]
I'm just getting started with fastapi-users and I've never used SQLAlchemy. My `db.py` matches the tutorial (https://github.com/fastapi-users/fastapi-users/blob/master/examples/sqlalchemy/app/db.py), specifically this class:
```python
class User(SQLAlchemyBaseUserTableUUID, Base):
    pass
```

I believe it's this class that is taken into account when the database is created and the table is generated. This is the structure of the `user` table:
![image](https://user-images.githubusercontent.com/11299982/176881251-f8ab2904-9ef4-4a5b-8f18-07126d32ce2c.png)

How can I change the class above so that instead of `email` there's `username` instead, and then also get rid of the `is_verified` field. I know the class references `SQLAlchemyBaseUSerTableUUID` which references `SQLAlchemyBaseUserTable` but just can't workout how to do this.

Any help appreciated, thanks!

**Comment by frankie567** [comment]
Hi @nickshanks347!

I'm afraid it's not possible to *remove* fields. You can always add new ones, but removing them would break FastAPI Users logic which rely on them.

**Reply by nickshanks347** [reply]
Thank you @frankie567. Just another quick question, I've copied your OAuth2 tutorial [here](https://fastapi-users.github.io/fastapi-users/10.1/configuration/oauth/?h=oauth2#sqlalchemy_1). Is there an easy way to integrate the returned `access_token` into the docs, so that if you've authenticated with Google then it will use that `access_token` for any protected routes?

Similarly, the route `/authenticated-route` depends on `current_active_user` which requires you to get an `access_token` by entering a username + password, this is fine. But when we bring OAuth2 into it, I can see my user account is generated when hitting the `/auth/google/callback` URL. How would I access the `authenticated-route` when using OAuth2? That depends on username + password for access token but I don't know the generated hashed_password because it used OAuth2.
